azavea/python-omgeo
# Feature/add mapzen search support

This adds support for Mapzen's search API, AKA Pelias. See the docs at https://mapzen.com/documentation/search/

Create an instance as follows:
```python
sources = OMGEO_SETTINGS = [[
    'omgeo.services.Mapzen',
    {
        'settings': {
            'api_key': 'search-XXXXXXX',
        },
    }
]]

g = Geocoder(sources=sources)
result = g.geocode(PlaceQuery(query='340 12th St, Philadelphia PA'))
```

A test file is attached and can be used to verify properly formatted results. 
[test.py.txt](https://github.com/azavea/python-omgeo/files/71192/test.py.txt)

### To Test
 * Get a mapzen search api key at https://mapzen.com/developers/sign_in
 * Install omgeo
 * Download the test file and update it with your API key
 * `python test.py` and watch for the results.
 * If you want to spin up a Pelias instance that can be done by following these instructions (https://github.com/pelias/vagrant)
 * Once working update test.py to point to your Pelias instance URL.
 * Try running the different PlaceQueries supplied and check the results. Using a key should result in a different endpoint call.
